### PR TITLE
avm2: FlexAssets

### PR DESCRIPTION
**Pull request by rdrpenguin04**
Tasks:
- [x] `IFlexAsset` stub
- [x] `ByteArrayAsset` stub
- [x] Functionality for `ByteArrayAsset` stub
- [ ] Possibly supporting other asset types

### CODE REVIEW

File: core/src/avm2/globals/mx/core/bytearrayasset.rs [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+use crate::avm2::class::{Class, ClassAttributes};
+use crate::avm2::method::Method;
+use crate::avm2::names::{Namespace, QName};
+use crate::avm2::object::Object;
+use crate::avm2::value::Value;
+use crate::avm2::Error;
+use gc_arena::{GcCell, MutationContext};
+
+/// Implements `mx.core.BitmapDataAsset`'s instance constructor.
+pub fn instance_init<'gc>(
```
Comment (adrian17) [inline core/src/avm2/globals/mx/core/bytearrayasset.rs:13]: (btw, these don't need to be `pub`, though for some reason a lot of these in existing code are)

File: core/src/avm2/globals/mx/core/iflexasset.rs [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+pub fn create_interface<'gc>(mc: MutationContext<'gc, '_>) -> GcCell<'gc, Class<'gc>> {
+    let class = Class::new(
+        QName::new(Namespace::package("mx.core"), "IFlexAsset"),
+        None,
+        Method::from_builtin(
+            bodiless_method,
+            "<IBitmapDrawable instance initializer>",
+            mc,
+        ),
+        Method::from_builtin(class_init, "<IFlexAsset interface initializer>", mc),
```
[the remark this answers is not in the document]
Comment (rdrpenguin04) [reply core/src/avm2/globals/mx/core/iflexasset.rs:40]: This was based on `IEventDispatcher` and `IBitmapDrawable`, which both say `<* interface initializer>`; should these all be changed?

File: core/src/avm2/globals/mx/core/iflexasset.rs [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+}
+
+/// Construct `IFlexAsset`'s class.
+pub fn create_interface<'gc>(mc: MutationContext<'gc, '_>) -> GcCell<'gc, Class<'gc>> {
+    let class = Class::new(
+        QName::new(Namespace::package("mx.core"), "IFlexAsset"),
+        None,
+        Method::from_builtin(
+            bodiless_method,
+            "<IBitmapDrawable instance initializer>",
```
[the remark this answers is not in the document]
Comment (rdrpenguin04) [reply core/src/avm2/globals/mx/core/iflexasset.rs:37]: Fixed

File: core/src/avm2/globals/mx/core/bytearrayasset.rs (hunk 1 above)
Comment (rdrpenguin04) [reply core/src/avm2/globals/mx/core/bytearrayasset.rs:13]: Okay; I'll keep it as `pub` until/unless the rest get fixed. This project may need some overall cleanup.

**Comment by rdrpenguin04** [comment]
It seems that `ByteArrayAsset` is primarily handled by the compiler; are there any other commonly-used Asset types I should find?

**Comment by adrian17** [comment]
Actually, wait, are we sure these are actually part of the engine, and not embedded in SWFs, like AVM1 mx stuff?
I just checked a random SWF and I see `mx.core.IFlexAsset` and `ByteArrayAsset` embedded in it.
(doesn't explain why I recall seeing them marked as not found in kmeishtax's errors report...)

**Comment by rdrpenguin04** [comment]
You're right; there may be something going wrong with parsing my test file. I'll close this and check a couple things.

**Pull request closed without merging**